### PR TITLE
Look up Administrator by SID on Windows

### LINKED ISSUE #52: admin_user hardcoded to "administrator" for Windows causes failures when administrator is renamed
In many environments, the administrator is renamed on Windows servers to something else for security reasons and this is a requirement for many security benchmarks.  After setting up Chef Automate, chef-client fails to run on any Windows server with a renamed administrator account due to the liveness agent being a required recipe and the inability to specify a different administrator account.

```
x.x.x.x [2018-04-23T14:08:48+00:00] FATAL: Chef::Exceptions::Win32APIError: directory[c:\chef\log\automate-liveness-agent] (c:/chef/cache/required_recipe line 935) had an error: Chef::Exceptions::Win32APIError: No mapping between account names and security IDs was done.

directory("c:\chef\log\automate-liveness-agent") do
       action [:create]
       default_guard_interpreter :default
       declared_type :directory
       recursive true
       owner "administrator"
       path "c:\\chef\\log\\automate-liveness-agent"
       mode nil
       group nil
     end 
```

### PR DESCRIPTION
**Pull request by btm**
The Windows Administrator user may be renamed, but it is a special account
so we can find it by SID.

Fixes #52

Signed-off-by: Bryan McLellan <btm@loftninjas.org>

### CODE REVIEW

File: lib/recipe.rb [hunk 1]
```diff
@@ -98,7 +98,12 @@
     solaris2
     suse
   )           => 'root',
-  %i(windows) => 'administrator'
+  %i(windows) =>  if platform?('windows')
+                    # This class is only loaded on Windows
```
Comment (markan) [inline lib/recipe.rb]: Out of curiosity, does the constant Chef::ReservedNames::Win32::Security::SID get evaluated if value_for_platform_family isn't windows? I would have thought that the platform? call wasn't needed, but always happy to learn.

Comment (btm) [reply lib/recipe.rb]: Yeah. I was disappointed it couldn't be that simple.

```
btm@oak:~$ /opt/chef/embedded/bin/irb
irb(main):001:0> require 'chef'
=> true
irb(main):002:0> admin_user = value_for_platform_family(
irb(main):003:1*   %i(
irb(main):004:1]     aix
irb(main):005:1]     amazon
irb(main):006:1]     debian
irb(main):007:1]     fedora
irb(main):008:1]     freebsd
irb(main):009:1]     mac_os_x
irb(main):010:1]     oracle
irb(main):011:1]     rhel
irb(main):012:1]     solaris2
irb(main):013:1]     suse
irb(main):014:1]   )           => 'root',
irb(main):015:1*   %i(windows) => Chef::ReservedNames::Win32::Security::SID.admin_account_name
irb(main):016:1> )
NameError: uninitialized constant Chef::ReservedNames::Win32::Security
Did you mean?  SecurityError
        from (irb):15
        from /opt/chef/embedded/bin/irb:11:in `<main>'
```

Comment (btm) [reply lib/recipe.rb]: simpler example:
```
chef:recipe (13.0.118)> platform?('windows')
 => false
chef:recipe (13.0.118)> admin_user = value_for_platform_family('windows' => Chef::ReservedNames::Win32::Security::SID.admin_account_name)
NameError: uninitialized constant Chef::ReservedNames::Win32::Security
        from (irb#1):56
```

**Comment by mdunc** [comment]
Any chance of this getting merged soon?  We've pretty much had to halt our conversion from our old Chef server to Opsworks/Chef Automate due to this.

**Comment by markan** [comment]
For anyone looking into this, the tests are failing on master. I'm currently poking at them now to see if I can get things running; once I do we can rebase this on top and merge.

**Comment by cynicaljoy** [comment]
Any updates?

**Comment by btm** [comment]
After rebasing on #54, `supported-win*-14` and `supported-win*-13` pass locally for me.

**Comment by btm** [comment]
```
  ✔  setup-001: converge previous stable version
     ✔  Command INTERVAL=2 chef-client -z -c /tmp/kitchen/test-client.rb -j /tmp/kitchen/test-stable-attrs.json exit_status should eq 0
  ✔  test-001: upgrade by converging the compiled recipe artifact
     ✔  Command INTERVAL=2 chef-client -z -c /tmp/kitchen/test-client.rb -j /tmp/kitchen/test-current-attrs.json exit_status should eq 0
  ✔  setup-002: reset the automate ping counter
     ✔  http GET on http://localhost:9292/organizations/ubuntu1604/reset-pings status should cmp == 200
     ✔  http GET on http://localhost:9292/organizations/ubuntu1604/reset-pings body should cmp == "0"
  ✔  setup-003: sleep to let the liveness agent send some pings
     ✔  Command sleep 30 exit_status should eq 0
  ∅  test-002: verify that the pings count has increased (1 failed)
     ∅  http GET on http://localhost:9292/organizations/ubuntu1604/pings body should not cmp == "0"
     
     expected it not to be == "0"
          got: 0
     
     (compared using `cmp` matcher)
```

I see a lot of this, where we run the liveness agent, then reset the count, wait, and expect there to have been some pings. But looking at the logs on the VM, the liveness agent is sleeping for 1800s, [which is the default](https://github.com/chef/automate-liveness-agent/blob/master/lib/automate_liveness_agent/liveness_update_sender.rb#L13).

While we do set `INTERVAL=2` when running the chef-client, we're ultimately running the liveness agent as a service, so that environment variable isn't going to stick. So those tests must have been broken for a while?

**Comment by btm** [comment]
the test-002 failure appears to be in places where `INTERVAL=2` isn't passed through the init service, here's a somewhat intrusive fix: https://github.com/chef/automate-liveness-agent/pull/55

**Comment by btm** [comment]
there's a lot of test suites. running them all effectively without system resources causing problems is super time consuming.

passed:
supported-ubuntu-*
supported-win-2016-*
supported-win-2016-std-12 took a second run
supported-sles-*
supported-solaris-*
supported-solaris-113 is hanging but supported-solaris-113-12 is passing. Might be a Chef 14 thing.

It'll take days to get through the rest of these.

**Pull request merged**